components-core/src/main/java/com/adyen/checkout/components/bundle/BundleHandler.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 28/11/2022.
+ */
+
+package com.adyen.checkout.components.bundle
+
+import androidx.annotation.RestrictTo
+import androidx.lifecycle.SavedStateHandle
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+/**
+ * Implement this in any view model or class that contains a [SavedStateHandle].
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface BundleHandler {
```

Comment:
This name doesn't feel right, because it's specifically for `SavedStateHandle`. `SavedStateHandler` would be better, but imo `SavedStateHandleContainer` would be even better. The implementing class is not really handling the `SavedStateHandle` it's just having one. Naming is hard, so feel free to suggest something else 😆

Reply:
You're right actually, I think the issue started in `DropInBundleHandler`. At first I only had this class and it handled the `SavedStateHandle` as well as the bundle. Then I decided to use the same approach for `SavedStateHandle` in the other parts of the project but forgot to rename it. Will fix 👍 

But as for the `DropInBundleHandler` class, do you think the name should change as well? This class is weird because it handles the `SavedStateHandle` inside it but also the bundle in its companion. I can't decide if it should be split in 2 classes or restructured/renamed 🤔

Reply:
That's a difficult one indeed 🤔 According to [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle) it should be split up in two. This would result in `class DropInSavedStateHandleContainer` and `DropInIntentCreator` (the first names that came to mind). If you put those two in the same file you can still share the constants and keep them private

Reply:
Yeah sharing the constants is what I was thinking of, I'll try to use one file 👍